### PR TITLE
Replace user mentions on account deletion

### PR DESCRIPTION
**Pull request by tyrandus**
#### Summary

This pull request provides a sample implementation of the proposed fix in [this issue](https://github.com/mattermost/mattermost-server/issues/20135). In short, when user accounts are deleted, all messages mentioning them via `@user` have the mention replaced by `@deletedUser`.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/20135

#### Release Note
```release-note
Added user mention anonymization for deleted user accounts.
```

### CODE REVIEW

**Comment by agnivade** [comment]
Thanks @tyrandus ! I like the idea but it needs some PM input. I'll ping on the issue.

**Comment by mkraft** [comment]
Screen Shot 2022-06-03 at 1 34 59 PM
Screen Shot 2022-06-03 at 1 34 43 PM

I was curious what Slack does so I made a data deletion request and this is how it's displayed in the UI.

**Review by mkraft — CHANGES_REQUESTED** [review]

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -994,6 +994,61 @@ func (s *SqlPostStore) PermanentDeleteByUser(userId string) error {
...
+func (s *SqlPostStore) ReplaceUserMentions(userId string) error {
+	transaction, err := s.GetMasterX().Beginx()
+	if err != nil {
+		return errors.Wrap(err, "begin_transaction")
+	}
+	defer finalizeTransactionX(transaction)
+
+	// get username
+	var username []string
+	err = transaction.Select(&username, "SELECT Username FROM Users WHERE Id = ? LIMIT 1", userId)
```
[the remark this answers is not in the document]
Comment (tyrandus) [reply store/sqlstore/post_store.go]: Good point.

File: app/user.go [hunk 2]
```diff
@@ -1551,6 +1551,10 @@ func (a *App) PermanentDeleteUser(c *request.Context, user *model.User) *model.A
 		return model.NewAppError("PermanentDeleteUser", "app.post.permanent_delete_by_user.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
+	if err := a.Srv().Store.Post().ReplaceUserMentions(user.Id); err != nil {
```
[the remark this answers is not in the document]
Comment (tyrandus) [reply app/user.go:1573]: True, but I am afraid that moving this to a background job is beyond my Go capabilities.

[the remark this answers is not in the document]
Comment (mkraft) [reply app/user.go:1573]: No worries. Thanks for the contribution, I'll link it for reference in a ticket to accomplish this.

[the remark this answers is not in the document]
Comment (mkraft) [reply app/user.go:1573]: https://mattermost.atlassian.net/browse/MM-45185

**Pull request closed without merging**